runtime/executor/src/main/java/edu/snu/nemo/runtime/executor/datatransfer/OutputCollectorImpl.java

```diff
@@ -74,12 +77,24 @@ public void clearMain() {
   }
 
   public void clearTag(final String tag) {
-    if (this.additionalTagElementsMap.get(tag) == null) {
+    if (this.mainTagOutputChildren.contains(tag)) {
       // This dstVertexId is for the main tag
       clearMain();
     } else {
       // Note that String#hashCode() can be cached, thus accessing additional output queues can be fast.
       this.additionalTagElementsMap.get(tag).clear();
     }
   }
+
+  public List<O> getMainTagOutputQueue() {
+    return mainTagElements;
+  }
+
+  public List getAdditionalTagOutputQueue(final String dstVertexId) {
+    if (this.mainTagOutputChildren.contains(dstVertexId)) {
+      return this.mainTagElements;
+    } else {
+      return this.additionalTagElementsMap.get(dstVertexId);
```

Comment:
throw exception when the returned value is null for sanity check?

Reply:
I will make these methods explicitly throw exceptions when wrong vertex id is passed.